storage: repeating error "error handling raft message: raft group deleted"

Found while manually testing a single-machine, six node cluster - one of the nodes encountered a condition which caused it to continuously spam the error "error handling raft message: raft group deleted".  

Here is the log on that node leading up to this point (node ID 3, store ID 2).

```
I160307 17:39:39.427088 storage/replica_command.go:1255  range 1: new leader lease replica {3 2 8} 2016-03-07 22:39:39 +0000 UTC +1.030s
I160307 17:39:43.394229 storage/replica_command.go:1255  range 21: new leader lease replica {3 2 11} 2016-03-07 22:39:40 +0000 UTC +3.973s
I160307 17:39:44.233112 server/status/runtime.go:137  runtime stats: 146 goroutines, 95.49MiB active, 42806.98cgo/sec, 0.63/0.18 %(u/s)time, 0.00 %gc (4x)
I160307 17:39:54.233116 server/status/runtime.go:137  runtime stats: 147 goroutines, 98.48MiB active, 40783.66cgo/sec, 0.52/0.16 %(u/s)time, 0.00 %gc (3x)
I160307 17:40:04.233253 server/status/runtime.go:137  runtime stats: 132 goroutines, 105.35MiB active, 40701.27cgo/sec, 0.50/0.15 %(u/s)time, 0.00 %gc (3x)
I160307 17:40:04.698772 storage/replica_command.go:1255  range 3: new leader lease replica {3 2 5} 2016-03-07 22:39:40 +0000 UTC +25.053s
I160307 17:40:14.233594 server/status/runtime.go:137  runtime stats: 133 goroutines, 63.40MiB active, 42124.36cgo/sec, 0.53/0.16 %(u/s)time, 0.00 %gc (4x)
I160307 17:40:24.233134 server/status/runtime.go:137  runtime stats: 132 goroutines, 72.37MiB active, 40963.88cgo/sec, 0.61/0.19 %(u/s)time, 0.00 %gc (3x)
I160307 17:40:34.233468 server/status/runtime.go:137  runtime stats: 132 goroutines, 81.48MiB active, 41730.67cgo/sec, 0.48/0.16 %(u/s)time, 0.00 %gc (3x)
I160307 17:40:43.409657 storage/replica_command.go:1255  range 21: new leader lease replica {3 2 11} 2016-03-07 22:40:15 +0000 UTC +28.977s
I160307 17:40:44.233867 server/status/runtime.go:137  runtime stats: 133 goroutines, 82.65MiB active, 40324.84cgo/sec, 0.47/0.15 %(u/s)time, 0.00 %gc (3x)
I160307 17:40:54.233526 server/status/runtime.go:137  runtime stats: 133 goroutines, 95.26MiB active, 42107.04cgo/sec, 0.48/0.16 %(u/s)time, 0.00 %gc (3x)
I160307 17:40:59.452146 storage/replica_command.go:1255  range 4: new leader lease replica {3 2 9} 2016-03-07 22:39:40 +0000 UTC +79.806s
I160307 17:40:59.453894 storage/replica.go:1072  raft: proposing ADD_REPLICA {6 6 12} for range 21
I160307 17:40:59.569226 storage/replica.go:1072  raft: proposing REMOVE_REPLICA {3 2 11} for range 21
I160307 17:40:59.569357 storage/replica.go:1072  raft: proposing REMOVE_REPLICA {3 2 11} for range 21
W160307 17:40:59.616705 storage/replica.go:1828  could not GC completed transaction: range 21 was not found
E160307 17:41:01.475186 storage/store.go:1817  error handling raft message: raft group deleted
E160307 17:41:01.740458 storage/store.go:1817  error handling raft message: raft group deleted
E160307 17:41:02.042060 storage/store.go:1817  error handling raft message: raft group deleted
E160307 17:41:02.341042 storage/store.go:1817  error handling raft message: raft group deleted
E160307 17:41:02.640434 storage/store.go:1817  error handling raft message: raft group deleted
```

The same message was spammed continuously until the server was stopped.

The duplication of the REMOVE_REPLICA command is somewhat suspicious; perhaps a replay which could never be completed, and caused some sort of infinite retry on the node?

None of the other five nodes reported anything suspicious, and according to the UI all ranges were fully replicated.  After stopping and restarting the cluster, the repeating message was no longer present.

--- a/storage/store.go
+++ b/storage/store.go
@@ -1754,8 +1754,19 @@ func (s *Store) handleRaftMessage(req *RaftMessageRequest) error {
 	// It deadlocks to hold s.Mutex while calling raftGroup.Step.
 	s.mu.Unlock()
 	if err != nil {
-		return util.Errorf("refusing incoming Raft message %s for group %d from %s to %s: %s",
-			req.Message.Type, req.GroupID, req.FromReplica, req.ToReplica, err)
+		// If getOrCreateReplicaLocked returns an error, log it at V(1)
+		// instead of returning it (where the caller will log it as
+		// Error). Errors here generally mean that the sender is out of
+		// date, and it may remain so until replicaGCQueue runs, so when
+		// these messages occur they will repeat many times.
+		// TODO(bdarnell): if we had a better way to report errors to the
+		// sender then we could prompt the sender to GC this replica
+		// immediately.
+		if log.V(1) {
+			log.Infof("refusing incoming Raft message %s for group %d from %s to %s: %s",
+				req.Message.Type, req.GroupID, req.FromReplica, req.ToReplica, err)
+		}
+		return nil
 	}
 	r.mu.Lock()
 	err = r.mu.raftGroup.Step(req.Message)

